dask/dataframe/_hyperloglog.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*
+u"""Implementation of HyperLogLog
+
+This implements the HyperLogLog algorithm for cardinality estimation, found
+in
+
+    Philippe Flajolet, Éric Fusy, Olivier Gandouet and Frédéric Meunier.
+        "HyperLogLog: the analysis of a near-optimal cardinality estimation
+        algorithm". 2007 Conference on Analysis of Algorithms. Nice, France
+        (2007)
+
+"""
+from __future__ import division
+
+import numpy as np
+import pandas as pd
+
+from .hashing import hash_pandas_object
+
+
+def compute_first_bit(a):
+    "Compute the position of the first nonzero bit for each int in an array."
+    # TODO: consider making this less memory-hungry
```

Comment:
It makes an (n, 32) array to answer a question about an (n,) array that shouldn't be so hard to do without any extra work.